### ISSUE TITLE
Missing libdns_sd.so.1

### ISSUE THREAD

**Issue by KirillChubarov** [mika314/obs-airplay#3]
OBS cant load plugin. Error: libdns_sd.so.1: cannot open shared object file: No such file or directory.
OS Ubuntu 22.04.1
OBS 27.1.3-modified (snap)

**Comment by mika314** [comment]
can you try to install libavahi-compat-libdnssd1
```
sudo apt install libavahi-compat-libdnssd1
```
I might miss it from the dependency list.

**Comment by mika314** [comment]
Actually, looking at my instructions, the dependency is listed. Did you run the following line?

```
sudo apt-get install -y clang pkg-config libavcodec-dev libavformat-dev libavutil-dev libswresample-dev git libobs-dev libavahi-compat-libdnssd-dev libplist-dev libfdk-aac-dev
```

**Comment by KirillChubarov** [comment]
Yes, i installed all dependencies from your instruction. Also try to install libavahi-compat-libdnssd1. It says already installed. Still have no luck, same error from OBS.

**Comment by KirillChubarov** [comment]
And i have libdns_sd.so.1 in my system:

>whereis libdns_sd.so.1
>libdns_sd.so.1: /usr/lib/x86_64-linux-gnu/libdns_sd.so.1

As i understand snap version of OBS cant use "normal" shared library located at /usr/lib/.... Shared libraries for snap apps must be located somewhere at /snap/core.... Looking for way to install shared library for snap apps.

**Comment by mika314** [comment]
Thank you for digging in.

This is why I am not happy with snap and flatpack: permission issues, sandboxing which suppose to "fix" dependency hell.

**Comment by KirillChubarov** [comment]
Can i build this plugin to static .so? To avoid using external system libraries.

**Comment by mika314** [comment]
The requirement for libdns_sd is coming from UxPlay tool (https://github.com/FDH2/UxPlay) you may ask them.
